src/main/haxe/yloader/impl/js/NodeLoader.hx

```diff
@@ -0,0 +1,176 @@
+package yloader.impl.js;
+
+import js.node.buffer.Buffer;
+import js.node.http.ClientRequest;
+import js.node.Http.HttpRequestOptions;
+import js.node.http.IncomingMessage;
+import js.node.Http;
+import js.node.Https.HttpsRequestOptions;
+import js.node.Https;
+import js.node.Url;
+import yloader.ILoader;
+import yloader.util.HeaderUtil;
+import yloader.util.StatusCodeUtil;
+import yloader.valueObject.Parameter;
+import yloader.valueObject.Request;
+import yloader.valueObject.Response;
+
+class NodeLoader implements ILoader
+{
+	public var onResponse:Response->Void;
+	public var request:Request;
+	public var responseEncoding = "utf-8";
+
+	var clientRequest:ClientRequest;
+	var response:IncomingMessage;
+	var responseString:String;
+
+	public function new(?request:Request)
+	{
+		this.request = request;
+	}
+
+	public function load():Void
+	{
+		cancel();
+		clientRequest = createClientRequest();
+		addClientRequestListeners();
+		writeData();
+		endRequest();
+	}
+
+	public function cancel():Void
+	{
+		if (clientRequest != null)
+		{
+			removeClientRequestListeners();
+			clientRequest.abort();
+		}
+		clientRequest = null;
+		response = null;
+		responseString = "";
+	}
+
+	function createClientRequest():ClientRequest
+	{
+		var url:UrlData = Url.parse(request.urlWithQuery, true);
+		switch (url.protocol)
+		{
+			case "http:":
+				return createHttpRequest(url);
+			case "https:":
+				return createHttpsRequest(url);
+			default:
+				return null;
+		}
+	}
+
+	function addClientRequestListeners():Void
+	{
+		clientRequest.on("error", onClientRequestError);
+	}
+
+	function removeClientRequestListeners():Void
+	{
+		clientRequest.removeListener("error", onClientRequestError);
+	}
+
+	function writeData():Void
+	{
+		if (requestHasData())
+			clientRequest.write(request.data);
+	}
+
+	function endRequest():Void
+	{
+		clientRequest.end();
+	}
+
+	function createHttpRequest(url:UrlData):ClientRequest
+	{
+		var options:HttpRequestOptions = createOptions(url);
+		return Http.request(options, onRequestResponse);
+	}
+
+	function createHttpsRequest(url:UrlData):ClientRequest
+	{
+		var options:HttpsRequestOptions = createOptions(url);
+		return Https.request(options, onRequestResponse);
+	}
+
+	function createOptions(url:UrlData):HttpsRequestOptions
+	{
+		var options:HttpsRequestOptions = {
+			host: url.host,
+			port: Std.parseInt(url.port),
+			path: url.path,
+			method: Std.string(request.method),
+			headers: getHeaders()
+		};
+
+		return options;
+	}
+
+	function getHeaders():Dynamic
+	{
+		var headers = HeaderUtil.toObject(request.headers);
+		setContentLengthHeader(headers);
+		return headers;
+	}
+
+	function setContentLengthHeader(headers:Dynamic):Void
+	{
+		if (!requestHasData()
+			|| Reflect.hasField(headers, "content-length")
```

Comment:
what about having `HeaderUtil.getContentLength` and `HeaderUtil.setContentLength`